docs/source/guide/s3.rst

```diff
@@ -5,7 +5,31 @@ S3
 
 By following this guide, you will learn how to use features of S3 client that
 are unique to the SDK, specifically the generation and use of pre-signed URLs,
-pre-signed POSTs, and the use of the transfer manager.
+pre-signed POSTs, and the use of the transfer manager. You will also learn how
+to use a few common, but important, settings specific to S3.
+
+
+Changing the Addressing Style
+-----------------------------
+
+S3 supports two different ways to address a bucket, Virtual Host Style and Path
+Style. This guide won't cover all the details of `virtual host addressing`_, but
+you can read up on that in S3's docs. In general, the SDK will handle the
+decision of what style to use for you, but there are some cases where you may
+want to set it yourself. For instance, if you have a CORS configured bucket
```

Comment:
We should add a note that when you do this you *have* to use the correct region for the bucket, whereas the default addressing style this is (typically) not required.